clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala

```diff
@@ -24,11 +23,24 @@ import org.broadinstitute.clio.util.model.{DataType, UpsertId}
 class ElasticsearchIndex[+CI <: ClioIndex](
   val clioIndex: CI,
   final val indexName: String,
+  final val orderedKeyFields: Seq[String],
   private[elasticsearch] val fieldMapper: ElasticsearchFieldMapper,
   private[elasticsearch] val defaultFields: Json = Json.obj()
 ) extends ModelAutoDerivation {
   import clioIndex.implicits._
 
+  private val keyFields = FieldMapper[clioIndex.KeyType].fields.map {
+    case (k, v) => ElasticsearchUtil.toElasticsearchName(k) -> v
```

Comment:
Why is this needed now that we have `orderedKeyFields`?

Reply:
It's needed to verify that `orderedKeyFields` contains all the required pieces. Will remove if we end up dropping that check from the main code.